### PR TITLE
feat(socket): Use OptionalValuePath in the udp socket config

### PR DESCRIPTION
**Pull request by spencergilbert**
Part of #15635

### CODE REVIEW

File: src/sources/socket/mod.rs [hunk 1]
```diff
@@ -279,35 +279,32 @@ impl SourceConfig for SocketConfig {
...
-                let port_key_path = config
+                let legacy_port_key = config
                     .port_key()
                     .as_ref()
-                    .and_then(|x| parse_value_path(x).ok())
-                    .map_or_else(
-                        || LegacyKey::InsertIfEmpty(owned_value_path!("port")),
-                        LegacyKey::InsertIfEmpty,
-                    );
+                    .map_or_else(|| parse_value_path("path").ok(), |k| k.path.clone())
```
Comment (neuronull) [inline src/sources/socket/mod.rs:294]: Should this be 

```suggestion
                    .map_or_else(|| parse_value_path("port").ok(), |k| k.path.clone())
```

File: src/sources/socket/udp.rs [hunk 2]
```diff
@@ -224,28 +227,28 @@ pub(super) fn udp(
...
                                         );
 
-                                        let port_key_path = config.port_key.as_ref().map_or_else(
-                                            || [BorrowedSegment::from("port")],
-                                            |key| [BorrowedSegment::from(key)],
-                                        );
+                                        let legacy_port_key = config
+                                            .port_key
+                                            .as_ref()
+                                            .map_or_else(|| parse_value_path("path").ok(), |k| k.path.clone());
```
Comment (neuronull) [inline src/sources/socket/udp.rs:246]: also here

Comment (spencergilbert) [reply src/sources/socket/udp.rs]: Fixed in [17c7645](https://github.com/vectordotdev/vector/pull/15666/commits/17c76453a258342a101933f934fe8c7c1af5401f)

File: src/sources/socket/mod.rs (hunk 1 above)
Comment (spencergilbert) [reply src/sources/socket/mod.rs]: Fixed in [17c7645](https://github.com/vectordotdev/vector/pull/15666/commits/17c76453a258342a101933f934fe8c7c1af5401f)

**Review by lukesteensen — APPROVED** [review]

**Review by neuronull — APPROVED** [review]

**Pull request merged**